sherpa/utils/testing.py

```diff
@@ -141,23 +143,16 @@ def decorator(test_function):
 
         return decorator
 
-    def requires_plotting(test_function):
```

Comment:
Hmmm. Pre this PR we had

- plot routines that could be tested even with no backend
- plot routines that required a backend
- plot routines that required a particular backend

With the loss of the chips backend, the latter two collapse into one (functinoally), but I think it makes sense for us to support these two different types of tests. I actually think that we should try and take all those tests that test matplotlib-specific functionality and place them into separate test files, rather than have them dotted through the place, and then have "I want a plotting backend but don't care what" tests in the general tests.

However, maybe there aren't really any tests that make sense for a generic "non-dummy" plot backend, in particular once we add a backend with a significantly-different behavior to chips/matplotlib.

Reply:
Two notes:

1.  Looking at this files by itself is somewhat misleading. While I removed `requires_plotting` I added two global fixtures that have pretty much the same use case (https://github.com/sherpa/sherpa/pull/1382/files#diff-2225729d3c21e6f344ee9cf2e165b679d8310a131c3ee462fd4e30f10aa14de2R553), while their implementation is somewhat different - they use the new `TemporaryPlottingBackend` context manager - and thus I found it easier to make a fixture then a decorator.
2. I see a slightly different list of cases:

    - plot tests that work with any backend (since we have the DummyBackend now, we'll never be in a situation where there is no backend). That's true for most plotting routines, yet, I want to argue that most tests should be in the next category:
    - plot tests that should work with all backends. In particular, the `IndepOnlyBackend` does not perform any plotting, but it does raise a warning when plotting options that are not on the list of backend-independent parameters are used. Users will never use this backend, but running (almost) every test with it ensures that we don't hardcode matplotlib-specific defaults into any of our objects. So, many, many tests are not run with the `all_plot_backends` fixtures. That increases the number of tests run dramatically and has some impact on the run time. If we had infinite CI, that would not be a problem, but since we don't we might scale that back in  a separate PR once all those changes and new backends are in.
    - plot tests that use backend specific code (e.g. `assert plt.gca.xlabel == "text"`). We have few of those and most already had a block of `if backend == pyplot: assert matplotlib specific stuff` so that most of the test is actually independent of th backend and it will pass without matplotlib.